core/src/main/kotlin/org/evomaster/core/search/gene/DoubleGene.kt

```diff
@@ -49,21 +61,39 @@ class DoubleGene(name: String,
             }
         }
 
-        //TODO min/max for Double
-        value = when (randomness.choose(listOf(0, 1, 2))) {
-            //for small changes
-            0 -> value + randomness.nextGaussian()
-            //for large jumps
-            1 -> value + (getDelta(randomness, apc) * randomness.nextGaussian())
-            //to reduce precision, ie chop off digits after the "."
-            2 -> BigDecimal(value).setScale(randomness.nextInt(15), RoundingMode.HALF_EVEN).toDouble()
-            else -> throw IllegalStateException("Regression bug")
+        /*
+           TODO min/max for double
+           Man: update a bit by considering min and max,
+           NEED a check by Andrea
+        */
+        var gaussianDelta = randomness.nextGaussian()
```

Comment:
make sure we have some unit tests for this, and the different branches are covered.

also, once computation is finished, you can have `assert` checks to verifiy the final results in within min/max

Reply:
there is a new unit test `NumberGeneTest.kt`. 

I will add e2e test.